versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/spi/KeyListBuildState.java

```diff
@@ -67,62 +60,50 @@ class KeyListBuildState {
     this.serializedEntrySize = serializedEntrySize;
   }
 
-  void add(KeyListEntry keyListEntry) {
-    int entrySize = serializedEntrySize.applyAsInt(keyListEntry);
-    if (embedded) {
-      // filling the embedded key-list in CommitLogEntry
-
-      if (currentSize + entrySize < maxEmbeddedKeyListSize) {
-        // CommitLogEntry.keyList still has room
-        addToEmbedded(keyListEntry, entrySize);
-      } else {
-        // CommitLogEntry.keyList is "full", switch to the first KeyListEntity
-        embedded = false;
-        newKeyListEntity();
-        addToKeyListEntity(keyListEntry, entrySize);
-      }
-    } else {
-      // filling linked key-lists via CommitLogEntry.keyListIds
-
-      if (currentSize + entrySize > maxKeyListEntitySize) {
-        // current KeyListEntity is "full", switch to a new one
-        finishKeyListEntity();
-        newKeyListEntity();
-      }
-      addToKeyListEntity(keyListEntry, entrySize);
-    }
+  void add(KeyListEntry entry) {
+    entries.add(entry);
   }
 
-  private void finishKeyListEntity() {
-    Hash id = randomHash();
-    newKeyListEntities.add(KeyListEntity.of(id, currentKeyList.build()));
-    newCommitEntry.addKeyListsIds(id);
+  int bucket(KeyListEntry entry, int bucketCount) {
+    return Math.abs(entry.getKey().hashCode() % bucketCount);
   }
 
-  private void newKeyListEntity() {
-    currentSize = 0;
-    currentKeyList = ImmutableKeyList.builder();
-  }
+  List<KeyListEntity> finish() {
+    int totalSize = entries.stream().mapToInt(serializedEntrySize).sum();
+    int remainingEmbedded = maxEmbeddedKeyListSize - currentSize;
```

Comment:
`currentSize` is the size of a `CommitLogEntry`, isn't it? Does it make sense to subtract it from the max "key list" size?